src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -1416,7 +1414,12 @@ bool SequentialSfMReconstructionEngine::Resection(const std::size_t viewIndex)
         const Vec2 xJ = _features_provider->feats_per_view.at(J)[track.at(J)].coords().cast<double>();
 
         // test if the track already exists in 3D
-        if (_sfm_data.structure.count(trackId) != 0)
+        bool trackIdExists;
+#pragma omp critical
```

Comment:
If two features are triangulated but do not meet some conditions (angle, rep error etc) their 3D point is not added (and the features are not removed from the track).
 
Now suppose that you have 2 views among the reconstructed views for which this is the case (they have a matching feature but not 3D point is generated because of those conditions).

If the current view being resectioned, `viewIndex`, has a feature matching with both those two features, we are exactly in this case: you have two views in `valid_views` that can potentially ask to reconstruct the point at the same time.